hook-producer/src/handlers/webhook.rs

```diff
@@ -30,8 +44,22 @@ pub async fn post(
         ));
     }
 
-    let url_hostname = get_hostname(&payload.url)?;
-    let job = NewJob::new(payload.max_attempts, payload, url_hostname.as_str());
+    let url_hostname = get_hostname(&payload.parameters.url)?;
+    // We could cast to i32, but this ensures we are not wrapping.
+    let max_attempts = i32::try_from(payload.max_attempts).map_err(|_| {
```

Comment:
To give more context: sqlx doesn't support encoding unsigned integer types to Postgres values. However, using the type system to enforce `max_attempts` are unsigned means we would never get bugs caused by a negative max attempt, which is great as negative max attempts makes no sense. So, we take an unsigned type, and then cast it to signed for sqlx support. Using `as` for casting could wrap around if we overflow, so we make the casting with `try_from` to fail on wrap around. I don't think we'll ever need more than `i32::MAX` max attempts anyways.

Perhaps a better solution would be to have a wrapper type that implements decode/encode over a unsigned integer, but this was easier to implement and apparently good enough for now.